app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args) -> Callable:
+        if args in cache_dict:
```

Comment:
Using `args` as a key in `cache_dict` can lead to a `TypeError` if the arguments contain mutable types like lists or dictionaries. Consider converting `args` to a hashable type, such as a tuple of tuples, to ensure all elements are hashable.